### PR TITLE
Add metrics to performance report + document in docs

### PR DESCRIPTION
**Pull request by jfomhover**
Updating the list of metrics reported by perf report at the end of the job for both single node and mpi multi node:
- added a couple more metrics to understand cpu utilization in detail
- renamed some of them for better precision
- creates documentation with an explicit definition.
- fix "disk write" metric (was using count instead of bytes)

### CODE REVIEW

File: docs/references/common/perf.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+The list of available metrics is detailed below:
+
+| Metric | Level | Description |
+| :-- | :-- | :-- |
+| `cpu_avg_utilization_over20_pct` | One value per node | How much time every cpu of that node are utilized more than 20%, over the total job time. |
+| `cpu_avg_utilization_over40_pct` | One value per node | How much time are all cpus are utilized more than 40%, over the total job time. |
+| `cpu_avg_utilization_over80_pct` | One value per node | ow much time are all cpus are utilized more than 80%, over the total job time. |
+| `cpu_avg_utilization_at100_pct` | One value per node | How much time are all cpus fully utilized at 100%, over the total job time. |
+| `cpu_avg_utilization_pct` | One value per node | How much are every cpu utilized on average during the entire job. |
+| `max_t_cpu_pct_per_cpu_avg` | One value per node | Maximum value taken by the **average cpu utilization** over the entire job time. |
```
Comment (ruizhuanguw) [inline docs/references/common/perf.md:12]: Does "Maximum value taken by XXX" mean "Maximum XXX"? If so, I feel the second might be more concise.

File: src/common/perf.py [hunk 2]
```diff
@@ -200,9 +200,36 @@ def report_nodes_perf(self):
...
+            )
+            self.metrics_logger.log_metric(
+                "cpu_avg_utilization_at100_pct",
+                sum( [ utilization >= 100.0 for utilization in cpu_avg_utilization])/len(cpu_avg_utilization)*100.0,
+                step=node
+            )
+            self.metrics_logger.log_metric(
+                "cpu_avg_utilization_over80_pct",
+                sum( [ utilization >= 80.0 for utilization in cpu_avg_utilization])/len(cpu_avg_utilization)*100.0,
+                step=node
```
Comment (ruizhuanguw) [inline src/common/perf.py:223]: Is there any way to make it clean to users about "step=node"?

Comment (jfomhover) [reply src/common/perf.py:223]: There's no way currently. Ideally, I'd want to store a table with 2 columns, one the metric value, second the node. But this is currently not possible with mlflow, so I'm using this trick to record the node id. We'll have to rely on documentation to help clarify.

File: docs/references/common/perf.md (hunk 1 above)
Comment (jfomhover) [reply docs/references/common/perf.md]: good suggestion, I applied.

**Review by ruizhuanguw — APPROVED** [review]

**Pull request merged**